natlib/svcd.c

```diff
@@ -185,3 +186,50 @@ static int svcd_init( lua_State *L )
 
     return 0;
 }
+
+static int svcd_subscribe(lua_State* L)
+{
+    if (lua_gettop(L) != 4) return luaL_error(L, "Expected (targetip, svcid, attrid, on_notify)");
+     
+    lua_getglobal(L, "SVCD");
+    lua_getfield(L, -1, "ivkid");
+     
+    uint16_t ivkid = luaL_checkinteger(L, -1);
```

Comment:
checkinteger is really only meant to be used for parameters. I think you probably want lua_tointeger, or lua_tonumber